src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        Storage.addShoppingCart(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> getById(Long shoppingCartId) {
+        return Storage.shoppingCarts.stream()
+                .filter(s -> s.getId().equals(shoppingCartId))
+                .findFirst();
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        return Storage.shoppingCarts;
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return Storage.shoppingCarts.stream()
+                .filter(s -> s.getUserId().equals(userId))
+                .findFirst()
+                .orElseThrow(() ->
+                        new NoSuchElementException("ShoppingCart of user id "
+                                + userId + " not found"));
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        IntStream.range(0, Storage.shoppingCarts.size())
+                .filter(i -> Storage.shoppingCarts.get(i).getId().equals(shoppingCart.getId()))
+                .forEach(i -> Storage.shoppingCarts.set(i, shoppingCart));
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean delete(ShoppingCart shoppingCart) {
+        return Storage.shoppingCarts.removeIf(cart -> cart.getId().equals(shoppingCart.getId()));
```

Comment:
RomanDubovskyi 15 hours ago
Suggested change 
        ```
return Storage.shoppingCarts
                .removeIf(shoppingCart1 -> shoppingCart.equals(shoppingCart1));
        return Storage.shoppingCarts.remove(shoppingCart);
```
if we don't delete by id, we can use remove() method it will work the same